server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java

```diff
@@ -320,7 +320,7 @@ private void addMessageIdNotFoundToResponse(MessageId messageId, SetMessagesResp
 
     private void handleInvalidRequest(SetMessagesResponse.Builder responseBuilder, MessageId messageId,
                                       List<ValidationResult> validationErrors) {
-        LOGGER.error("Invalid update request for message #{}", messageId);
+        LOGGER.warn("Invalid update request for message #{}: {}", messageId, validationErrors);
```

Comment:
not sure putting a list in a log will work, every List implementation don't have a good toString.
We should write `validationsErrors.stream().collect(Collectors.join(", "))` or something like that.

Reply:
Only immutable lists are passed to that method. Immutable lists have good rendering. I would be in favor of enforcing an immutable list as argument.